core/src/main/java/macrobase/analysis/summary/itemset/RiskRatio.java

```diff
@@ -0,0 +1,62 @@
+package macrobase.analysis.summary.itemset;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RiskRatio {
+    private static final Logger log = LoggerFactory.getLogger(RiskRatio.class);
+
+    private static double computeDouble(double exposedInlierCount,
+                                        double exposedOutlierCount,
+                                        double totalInliers,
+                                        double totalOutliers) {
+        double totalExposedCount = exposedInlierCount + exposedOutlierCount;
+        double unexposedOutlierCount = (totalOutliers - exposedOutlierCount);
+        double totalMinusExposedCount = totalInliers + totalOutliers - totalExposedCount;
+
+        // no exposure occurred
+        if (totalExposedCount == 0) {
+            log.error("Computing risk ratio with no exposure.");
+            return 0;
+        }
+
+        // we only exposed this ratio, everything matched!
+        if (totalMinusExposedCount == 0) {
+            return 0;
```

Comment:
Let's discuss this tomorrow. Our thinking was that, if everything is exposed, it's not clear the utility the RR will have; specifically, in the case of MB queries, it likely means you misconfigured your query. Compared to adding 0.5 ([also](http://handbook.cochrane.org/chapter_16/16_9_2_studies_with_zero_cell_counts.htm)), we thought returning zero given the expected use cases would be more intuitive. But maybe we should just `log.warn` and do the 0.5 trick.